### PR TITLE
NGP-2426: adding in site mapper to retrieve all urls not intially tak…

### PR DESCRIPTION
**Pull request by tsmithwirewheel**
…en by normal searching.

### CODE REVIEW

File: src/SiteMapperService.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    logger.error(e, `Not able to retrieve list of sites for:  ${url}`)
+    return []
+  }
+}
+
+const makeFullSiteMapUrlString = (url: string): string => {
+  return `${url}/sitemap.xml`
+}
+
+const convertToLinksForBlackLight = (sites: string[]): any[] => {
```
Comment (ghost) [inline src/SiteMapperService.ts]:
```suggestion
const convertToLinksForBlackLight = (sites: string[]) => {
```

We should remove the any and either allow typescript to infer the type, or we should explicitly define the return type.

Comment (tsmithwirewheel) [reply src/SiteMapperService.ts]: Implemented

File: src/collector.ts [hunk 2]
```diff
@@ -221,8 +222,12 @@ export const collector = async ({
 
   output.uri_dest = page.url();
   duplicatedLinks = await getLinks(page);
+
+  const sitemapperSites = await retrieveSites(page.url())
+  let joinedDuplicatedLinks = [...duplicatedLinks, ...sitemapperSites]
```
Comment (ghost) [inline src/collector.ts]: What's the purpose of creating a new variable here? Couldn't you just reassign the `duplicatedLinks` again?

```suggestion
  duplicatedLinks = await getLinks(page);

  const sitemapperSites = await retrieveSites(page.url())
  duplicatedLinks = [...duplicatedLinks, ...sitemapperSites]
```

Or maybe even

```suggestion
  duplicatedLinks = [...(await getLinks(page)), ...(await retrieveSites(page.url()))]
```

Or possibly 

```suggestion
  duplicatedLinks = (await getLinks(page)).concat((await retrieveSites(page.url())))
```

Comment (tsmithwirewheel) [reply src/collector.ts]: I am trying to leave the old code in place and make it clear what I am adding in.  There is always ways to do multiple things.  The last way you suggested definitely reduces lines of code.

File: src/SiteMapperService.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    logger.error(e, `Not able to retrieve list of sites for:  ${url}`)
+    return []
+  }
+}
+
+const makeFullSiteMapUrlString = (url: string): string => {
+  return `${url}/sitemap.xml`
+}
+
+const convertToLinksForBlackLight = (sites: string[]) => {
```
Comment (jay-delanoy-wirewheel) [inline src/SiteMapperService.ts]:
```suggestion
const convertToLinksForBlackLight = (sites: string[]): { href: string } => {
```

File: src/SiteMapperService.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import {getLogger} from "./logger";
+import {join} from "path";
+
+const outDir = join(process.cwd(), "bl-tmp")
+const quiet = true
+const logger = getLogger({ outDir, quiet });
+
+
+
+export const retrieveSites = async (url: string): Promise<any[]> => {
```
Comment (jay-delanoy-wirewheel) [inline src/SiteMapperService.ts]: Isn't this type more accurate?

```suggestion
export const retrieveSites = async (url: string): Promise<{href: string}[]> => {
```

File: src/collector.ts (hunk 2 above)
Comment (jay-delanoy-wirewheel) [reply src/collector.ts]: I hate re-assigning so generally argue against it.

Also, in this particular case, the `let` can be a `const` -- while the array is being mutated in line 275, there doesn't seem to be any reason to do a concat/reassign vs `push`ing the new links in.

File: src/SiteMapperService.ts (hunk 3 above)
Comment (jay-delanoy-wirewheel) [reply src/SiteMapperService.ts]: My bad, 
```suggestion
const convertToLinksForBlackLight = (sites: string[]): { href: string }[] => {
```

**Pull request merged**